src/main/java/com/amazon/opendistroforelasticsearch/security/user/User.java

```diff
@@ -264,7 +264,10 @@ public final void addOpenDistroSecurityRoles(final Collection<String> securityRo
         return this.openDistroSecurityRoles == null ? Collections.emptySet() : Collections.unmodifiableSet(this.openDistroSecurityRoles);
     }
 
-    public final String getUserRolesString() {
-        return name + "|" + String.join(",", getRoles()) + "|" + String.join(",", getOpenDistroSecurityRoles());
+    public final String getUserRolesTenantString() {
+        return name + "|"
+                + String.join(",", getRoles()) + "|"
+                + String.join(",", getOpenDistroSecurityRoles()) + "|"
+                + requestedTenant;
```

Comment:
Can `requestedTenant` be `null`? Will it be better to skip `requestedTenant` if it is not set?

Reply:
Yes it can be null and is handled in common-utils. however I like your suggestion and updated to add requestedTenant only if non-null.